core/src/test/java/com/google/googlejavaformat/java/CommandLineOptionsParserTest.java

```diff
@@ -152,4 +160,12 @@ public void illegalLines() {
     }
   }
 
+  @Test
+  public void sourcesFile() throws IOException {
+    Path path = testFolder.newFile("HelloAndGoodbye.txt").toPath();
+    Files.write(path, "Hello.java\n   \n\nℕ.java\nGoodbye.java".getBytes(UTF_8));
+    CommandLineOptions options =
+        CommandLineOptionsParser.parse(Arrays.asList("@" + path.toString()));
+    assertThat(options.files()).containsExactly("Hello.java", "ℕ.java", "Goodbye.java");
+  }
```

Comment:
Can you also test non-params file arguments before/after the params file, the `@@`-escape syntax, and maybe a nested params file?

Reply:
Sure.